internal/pkg/cli/env_delete.go

```diff
@@ -346,6 +369,78 @@ func (o *deleteEnvOpts) ensureRolesAreRetained() error {
 	return nil
 }
 
+// emptyBuckets returns nil if buckets were deleted successfully. Otherwise, returns the error.
+func (o *deleteEnvOpts) emptyBuckets() error {
+	s3buckets, err := o.rg.GetResources(&resourcegroupstaggingapi.GetResourcesInput{
+		ResourceTypeFilters: aws.StringSlice([]string{"s3:bucket"}),
+		TagFilters: []*resourcegroupstaggingapi.TagFilter{
+			{
+				Key:    aws.String(stack.EnvS3BucketStackNameTagKey),
+				Values: []*string{aws.String(stack.NameForEnv(o.appName, o.name))},
+			},
+			{
+				Key:    aws.String(stack.EnvS3BucketLogicalIDTagKey),
+				Values: aws.StringSlice(stack.EnvManagedS3BucketLogicalIds),
+			},
+			{
+				Key:    aws.String(deploy.EnvTagKey),
+				Values: []*string{aws.String(o.name)},
+			},
+			{
+				Key:    aws.String(deploy.AppTagKey),
+				Values: []*string{aws.String(o.appName)},
+			},
+		},
+	})
+
+	if err != nil {
+		return fmt.Errorf("find s3 bucket resources: %w", err)
+	}
+
+	envResources, err := o.envStackDescriber.Resources()
+	if err != nil {
+		return fmt.Errorf("find stack resources: %w", err)
+	}
+
+	var stackBucketARNs []string
+	for _, resource := range envResources {
+		if contains(resource.PhysicalID, stack.EnvManagedS3BucketLogicalIds) {
+			stackBucketARNs = append(stackBucketARNs, resource.PhysicalID)
+		}
+	}
+
+	var failedBuckets []string
+	var bucketErrors []error
+	for _, resourceTagMapping := range s3buckets.ResourceTagMappingList {
+		bucketARN, err := arn.Parse(aws.StringValue(resourceTagMapping.ResourceARN))
+		if err != nil {
+			return fmt.Errorf("parse the arn %s: %w", aws.StringValue(resourceTagMapping.ResourceARN), err)
+		}
+
+		// Attempt to empty all buckets found via GetResources API call
+		if err = o.s3.EmptyBucket(bucketARN.Resource); err != nil {
+			failedBuckets = append(failedBuckets, bucketARN.Resource)
+			bucketErrors = append(bucketErrors, err)
+			continue
+		}
+
+		// Warn about hanging bucket when bucket is found via API call but is not in the env CFN stack
+		// Those found via API call but not CFN stack cannot be deleted by Copilot
+		if !contains(bucketARN.String(), stackBucketARNs) {
+			log.Warningln(`Bucket %q was emptied, but was not found in the Cloudformation stack. This resource is now dangling, and can be deleted from the S3 console.`, bucketARN)
```

Comment:
nit:
```suggestion
			log.Warningln(`Bucket %q was emptied, but was not associated with the environment Cloudformation stack. This resource is now dangling, and can be deleted from the S3 console.`, bucketARN)
```

Reply:
Is "associated with" the typical wording you use when talking about resources created by a cloudformation stack? I think this could be misleading because a user might think that "associated with" just means that it's related to the stack, which I think some people may still consider a dangling resource to be associated with the stack. I think I'd rather use language that says that the bucket is no longer defined in the stack. What are your thoughts on this wording?